src/main.rs

```diff
@@ -0,0 +1,289 @@
+//! STAR Randomness web service
+
+use axum::extract::{Json, State};
+use axum ::http::StatusCode;
+use axum::{routing::get, routing::post, Router};
+use base64::prelude::{Engine as _, BASE64_STANDARD as BASE64};
+use serde::{Deserialize, Serialize};
+use tracing::{debug, info};
+
+use ppoprf::ppoprf;
+use std::sync::{Arc, RwLock};
+
+
+/// Internal state of the OPRF service
+struct OPRFServer {
+    /// oprf implementation
+    server: ppoprf::Server,
+    /// currently-valid randomness epoch
+    epoch: u8,
+}
+
+/// Shareable wrapper around the server state
+type OPRFState = Arc<RwLock<OPRFServer>>;
+/// Represents an contiguous range of configured randomness epochs
+type EpochRange = std::ops::Range<u8>;
+
+impl OPRFServer {
+    /// Initialize a new OPRFServer state supporting the given list of epochs
+    fn new(epochs: &EpochRange) -> Result<Self, ppoprf::PPRFError> {
+        // ppoprf wants a vector, so generate one from our range.
+        let epochs: Vec<u8> = epochs.to_owned().collect();
+        let epoch = epochs[0];
+        let server = ppoprf::Server::new(epochs)?;
+        Ok(OPRFServer{ server, epoch })
+    }
+}
+
+
+/// Request format for the randomness endpoint
+#[derive(Deserialize, Debug)]
+struct RandomnessRequest {
+    /// Array of points to evaluate
+    /// Should be base64-encoded compressed Ristretto curve points
+    points: Vec<String>,
+    /// Optional request for evaluation within a specific epoch
+    epoch: Option<u8>,
+}
+
+/// Response format for the randomness endpoint
+#[derive(Serialize, Debug)]
+struct RandomnessResponse {
+    /// Resulting points from the OPRF valuation
+    /// Should be base64-encode compressed points in one-to-one
+    /// correspondence with the request points.
+    points: Vec<String>,
+    /// Randomness epoch used in the evaluation.
+    epoch: u8,
+}
+
+/// Maximum number of points acceptable in a single request
+const MAX_POINTS: usize = 1024;
```

Comment:
This seems worth documenting somewhere, even just in our P3A Google doc, just in case we ever run into that limit.

Reply:
Ok. I added a note to the Advanced P3A google doc. This limit is also advertised by the server on the `/info` endpoint, alongside the current epoch and keys for ZK verification.